speaker/layer2_controller.go

```diff
@@ -223,8 +223,12 @@ func poolMatchesNodeL2(pool *config.Pool, node string) bool {
 }
 
 func (c *layer2Controller) speakersForPool(pool *config.Pool, nodes map[string]*v1.Node) map[string]bool {
+	usableSpeakers := c.sList.UsableSpeakers()
+	if usableSpeakers == nil {
+		return nil
```

Comment:
here I would be more mindful and IF we want to match the previous logic, we should assume that if member list is not available (which is technically different from usableSpeakers returning nil), we iterate over all the nodes.

This has a very important flaw though: metallb will behave as if all the nodes have a running speaker (which is something I am not sure we really want).

So, I would say that if we want to fix this we need to take into account looking where the speakers are running in order to avoid to elect one node without a speaker.